csrc/multidevice/collective.h

```diff
@@ -0,0 +1,172 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#ifdef USE_DISTRIBUTED
+
+#include <multidevice/communicator.h>
+#include <multidevice/multidevice.h>
+
+namespace nvfuser {
+
+/*
+  This struct gathers all the parameters necessary for the
+  construction a collective
+*/
+struct TORCH_CUDA_CU_API CollectiveParams {
+  DeviceIdxType root = -1;
```

Comment:
Shouldn't this be `RankType`?

Reply:
no, see https://github.com/NVIDIA/Fuser/pull/797#discussion_r1315053784